server/src/main/java/com/testsigma/service/TestStepService.java

```diff
@@ -574,8 +577,40 @@ private List<ActionTestDataMap> getMapsList(){
 
         return actionsMap;
     }
+    public void bulkDelete(Long[] ids) throws ResourceNotFoundException {
+        for (Long id:ids){
+              this.destroy(this.find(id));
+            }
+    }
 
-
+    private Boolean checkBulkDeleteStepsHaveAllPrerequisiteSteps(List<Long> ids,List<TestStep> testStepList){
+        for (TestStep testStep:testStepList){
+            if(!ids.contains(testStep.getId())){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public List<TestStepDTO> checkForLinkedTestCases(Long[] ids) throws ResourceNotFoundException {
+        Long testCaseId = this.find(ids[0]).getTestCaseId();
+        List<TestStepDTO> testSteps = new ArrayList<>();
+        List<Long> idsList=Arrays.asList(ids);
+        int index=0;
+        for(Long id:ids) {
+            List<TestStep> testStepList= this.repository.findAllByTestCaseIdAndPreRequisiteStepId(testCaseId,id);
+            if(!testStepList.isEmpty() && !checkBulkDeleteStepsHaveAllPrerequisiteSteps(idsList,testStepList)){
```

Comment:
* So this method doesn't return the prerequisite test steps if it is already part of the ids. That would be misleading. There is no way the user will know this while checking for linked test steps as part of the controller action. I would suggest you to include all the prerequisite test steps for the supplied ids. You should do the filtering as part of the frontend to choose which prerequisites you want to use. 

* Also, the user should know which prerequisite test steps correspond to which ids. Here, the user cant find that out if you simply return the list. Use something like a map or create a separate dto

Reply:
1) Rajesh said if user is deleting a step 
example step1 is prerequisite for step 2 & 3  , if user is deleting step 1,2,3 then we need not to show those steps.
if user is deleting only 1,2 steps then we need to give alert .

2)This was done by adding steps in a order and displaying them an understandable way.
![image](https://user-images.githubusercontent.com/106070370/197157143-544dadc9-f597-4066-bb0a-60f772825fb9.png)

Reply:
In my comments, I am not talking about how/what you want to show the user. I am talking about the response expected from the functions itself. If you have a controller action, you should ensure that it can be reused for any other usecases as well. Once you return the expected response from backend, you can filter based on your specific usecase. In this case, as Rajesh suggested, have the same view but rather than filtering it in backend for your specific usecase, return all the linked test steps and then based on your usecase, filter what you want to show